From 8c1cd7c6d838264abd06e670633a1a060d9cc9f3 Mon Sep 17 00:00:00 2001
From: yunwei37 <yunwei356@gmail.com>
Date: Mon, 4 Mar 2024 07:03:16 +0000
Subject: [PATCH] fix for 23

---
 app/pktgen-cmds.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/app/pktgen-cmds.c b/app/pktgen-cmds.c
index 9f46fc1..cddd223 100644
--- a/app/pktgen-cmds.c
+++ b/app/pktgen-cmds.c
@@ -1945,14 +1945,14 @@ enable_bonding(port_info_t *info, uint32_t state)
         return;
     }
 
-    num_workers = rte_eth_bond_members_get(info->pid, workers, RTE_MAX_ETHPORTS);
+    num_workers = rte_eth_bond_slaves_get(info->pid, workers, RTE_MAX_ETHPORTS);
     if (num_workers < 0) {
         printf("Failed to get worker list for port = %d\n", info->pid);
         return;
     }
 
     num_active_workers =
-        rte_eth_bond_active_members_get(info->pid, active_workers, RTE_MAX_ETHPORTS);
+        rte_eth_bond_active_slaves_get(info->pid, active_workers, RTE_MAX_ETHPORTS);
     if (num_active_workers < 0) {
         printf("Failed to get active worker list for port = %d\n", info->pid);
         return;
@@ -2050,7 +2050,7 @@ show_bonding_mode(port_info_t *info)
         printf("\n");
     }
 
-    num_workers = rte_eth_bond_members_get(port_id, workers, RTE_MAX_ETHPORTS);
+    num_workers = rte_eth_bond_slaves_get(port_id, workers, RTE_MAX_ETHPORTS);
 
     if (num_workers < 0) {
         printf("\tFailed to get worker list for port = %d\n", port_id);
@@ -2066,7 +2066,7 @@ show_bonding_mode(port_info_t *info)
         printf("\tSlaves: []\n");
     }
 
-    num_active_workers = rte_eth_bond_active_members_get(port_id, workers, RTE_MAX_ETHPORTS);
+    num_active_workers = rte_eth_bond_active_slaves_get(port_id, workers, RTE_MAX_ETHPORTS);
 
     if (num_active_workers < 0) {
         printf("\tFailed to get active worker list for port = %d\n", port_id);
@@ -2084,10 +2084,10 @@ show_bonding_mode(port_info_t *info)
     }
 
     for (i = 0; i < num_active_workers; i++) {
-        struct rte_eth_bond_8023ad_member_info conf;
+        struct rte_eth_bond_8023ad_slave_info conf;
 
         printf("\t\tSlave %u\n", workers[i]);
-        rte_eth_bond_8023ad_member_info(info->pid, workers[i], &conf);
+        rte_eth_bond_8023ad_slave_info(info->pid, workers[i], &conf);
         printf("\t\t  %sSelected\n\t\t  Actor States  ( ", conf.selected ? "" : "Not ");
         show_states(conf.actor_state);
         printf(")\n\t\t  Partner States( ");
-- 
2.40.1

